### DISCUSSION TITLE
How to set layout order of child nodes?

### DISCUSSION THREAD

**Discussion by bramarcade** [xyflow/xyflow#3208] [Q&A]
I'm making a game dialogue editor, meaning that some nodes have multiple child nodes as responses to a piece of dialogue.
................................ Response Node A
.............................. /
Dialogue Node -Response Node B
............................... \\
................................ Response Node C

I want to be able to edit the vertical order of Response Nodes A/B/C, such that the order from top to bottom can be made C/B/A or A/C/B or any order. This is crucial for the editor as the user might write a line and then decide that it should be presented first, or last, or in the middle. I can just add data for the game to pick up on, but I'd like the visual order to be reflected in the editor itself if possible. What's the best way to go about this?

**Comment by bcakmakoglu** [comment]
Are you _sure_ you need *child* nodes for this? 
In a lot of these cases a simple list inside a node would probably be sufficient, add some draggable library to it so you can change the list order by dragging and you're good to go, no need to use nested nodes.

A library that could be used to create a sortable list is [`dnd-kit`](https://docs.dndkit.com/presets/sortable).

**Comment by bramarcade** [comment]
I'm taking inspiration from Articy: Draft, which uses smaller composable nodes over larger configurable ones.

**Reply by bcakmakoglu** [reply]
Mh... I see, maybe I misunderstood what you meant by child nodes. I am under the assumption that you referred to them as like in [Sub Flows (like in this example)](https://reactflow.dev/docs/examples/layout/sub-flows/).

From what I've seen after taking a quick look at a video about Articy, it doesn't look like it uses any sort of Sub Flows.
So when you refer to child nodes are you just referring to a node that comes after another one? 😄

**Reply by bramarcade** [reply]
Oh, yes, just the nodes coming after another node. My mistake.

**Reply by bcakmakoglu** [reply]
In that case, a draggable solution could be tricky - how would you go about switching node positions? 🤔

Kinda think it would be easier to solve this with what I suggested basically, one node that contains all dialog options in a list and you can just modify the lists sort order to change where each dialog option is.

**Reply by bramarcade** [reply]
I'd switch node positions by having up and down arrow icons on them. Clicking an up arrow would move the node further up. I would do this onClick by reordering the relevant nodes using `getNodes` and `setNodes` as provided by `useReactFlow`.

I guess my refined question is: is the underlying node array order the only thing determining which of nodes A/B/C get put on top and which get put on bottom? Or is there something other than node order I would have to change to control which sibling got put on top?

**Reply by bramarcade** [reply]
Ah, never mind, I think the functionality I'm referring to is part of the d3-layout stuff in the pro examples. I'll take a look there, thanks.

**Reply by bcakmakoglu** [reply]
Ah that makes sense 😅
About questions regarding pro examples, you should probably send a mail to `info@reactflow.dev`
